TaylorSource/CoreData/NSFRCDatasource.swift

```diff
@@ -0,0 +1,80 @@
+import UIKit
+import CoreData
+
+public struct NSFRCDatasource<
+    Factory
+    where
+    Factory: _FactoryType,
+    Factory.ViewType: NSFRCIndexedUpdateConsumer,
+    Factory.CellIndexType == NSFRCCellIndex,
+Factory.SupplementaryIndexType == NSFRCSupplementaryIndex>: DatasourceType {
+    
+    public typealias FactoryType = Factory
+    
+    public let identifier: String
+    public let factory: Factory
+    public var title: String? = .None
+    
+    public let updateHandler = NSFRCUpdateHandler()
+    public var selectionManager = IndexPathSelectionManager()
+    
+    public init(id: String, fetchedResultsController: NSFetchedResultsController, factory f: Factory) {
+        fetchedResultsController.delegate = self.updateHandler
+        self.fetchedResultsController = fetchedResultsController
+        identifier = id
+        factory = f
+    }
+    
+    public var numberOfSections: Int {
+        return self.fetchedResultsController.sections?.count ?? 0
+    }
+    
+    public func numberOfItemsInSection(sectionIndex: Int) -> Int {
+        if let section = self.fetchedResultsController.sections?[sectionIndex] {
+            return section.numberOfObjects
+        }
+        return 0
```

Comment:
One of the things I'm (albeit very slowly) working on with TaylorSource is an update to more Swift 2.0 style code. So for things like this, where there is a "good path" and a "bad path", I think it would be better to use `guard` with an early return statement. For example:

```swift
guard let section = fetchedResultsController.sections?[sectionIndex] else { return 0 }
return section.numberOfObjects
```

Same philosophy applies to most of these methods here. Appreciate that this is the style of the rest of TaylorSource at the moment... 😔